assets/js/pages/SapSystemDetails/GenericSystemDetails.jsx

```diff
@@ -30,28 +35,46 @@ const getUniqueHosts = (hosts) =>
       .values()
   );
 
+// it includes SAP and HANA instance operations
+const instanceStartStopOperations = [SAP_INSTANCE_START, SAP_INSTANCE_STOP];
+
+const modalInitialState = { open: false, operation: '' };
+
+const closeInstanceModal = (prevState) => ({ ...prevState, open: false });
+
 export function GenericSystemDetails({
   title,
   type,
   system,
   userAbilities,
   cleanUpPermittedFor,
   onInstanceCleanUp,
+  operationsEnabled = false,
+  getInstanceOperations = noop,
 }) {
   if (!system) {
     return <div>Not Found</div>;
   }
   const [cleanUpModalOpen, setCleanUpModalOpen] = useState(false);
   const [instanceToDeregister, setInstanceToDeregister] = useState(undefined);
+  const [operationModalOpen, setOperationModelOpen] =
+    useState(modalInitialState);
+  const [currentOperationInstance, setCurrentOperationInstance] =
+    useState(undefined);
 
   const onCleanUpClick = (instance) => {
     setCleanUpModalOpen(true);
     setInstanceToDeregister(instance);
   };
 
+  const curriedGetInstanceOperations = getInstanceOperations(
+    null, // placeholder for runningOperations to be used at some point
```

Comment:
Just wondering in which direction will this placeholder go in the future ? 
What will it do ?

Reply:
I will pass the `runningOperations` parameter to set operations in running state.
I wanted to allocate the place, as `curry` function uses the paremeter in the order that are given, and I wanted to have the `runningOperations` first.
You will see in the next PR hehe